analytic_plan_mass_create/wizard/analytic_plan_mass_create.py

```diff
@@ -1,113 +1,98 @@
-        }]
-
-    def default_get(self, cr, uid, fields, context=None):
-        res = super(AnalyticPlanMassCreate, self).default_get(
-            cr, uid, fields, context=context)
-        analytic_obj = self.pool['account.analytic.account']
-        analytic_account_ids = context.get('active_ids', [])
-        active_model = context.get('active_model')
+            'date': time.strftime('%Y-%m-%d'),
+            'labor_cost': 0.0
+        }
 
+    @api.model
+    def default_get(self, fields):
+        res = super(AnalyticPlanMassCreate, self).default_get(fields)
+        analytic_obj = self.env['account.analytic.account']
+        analytic_account_ids = self._context.get('active_ids', [])
+        active_model = self._context.get('active_model')
         if not analytic_account_ids:
             return res
         assert active_model == 'account.analytic.account', \
             'Bad context propagation'
-
         items = []
-        for account in analytic_obj.browse(cr, uid, analytic_account_ids,
-                                           context=context):
-                items += self._prepare_item(cr, uid, account, context=context)
+        for account in analytic_obj.browse(analytic_account_ids):
+            items.append((0, 0, self._prepare_item(account)))
         res['item_ids'] = items
-
         return res
 
-    def _prepare_analytic_line_plan_common(self, cr, uid, wizard, item,
-                                           context=None):
-
+    @api.model
+    def _prepare_analytic_line_plan_common(self, wizard, item):
         return {
             'account_id': item.account_id.id,
             'name': item.account_id.name,
             'date': item.date,
             'currency_id': wizard.template_id.currency_id.id,
-            'user_id': uid,
+            'user_id': self._uid,
             'company_id': item.account_id.company_id.id,
             'version_id': wizard.template_id.version_id.id
         }
 
-    def _prepare_analytic_line_plan(self, cr, uid, wizard, item, product,
-                                    amount_currency, type, common,
-                                    context=None):
-        plan_line_obj = self.pool['account.analytic.line.plan']
-        am = plan_line_obj.on_change_amount_currency(
-            cr, uid, False, amount_currency,
-            wizard.template_id.currency_id.id,
-            item.account_id.company_id.id, context=context)
-        if am and 'value' in am and 'amount' in am['value']:
-            amount = am['value']['amount']
+    @api.multi
+    def _prepare_analytic_line_plan(self, wizard, item, product,
+                                    amount_currency, type, common):
+        analytic_line_plan_obj = self.env['account.analytic.line.plan']
+        line_ids = analytic_line_plan_obj.\
+            search([('account_id', '=', item.account_id.id),
+                    ('version_id', '=', wizard.template_id.version_id.id)])
+        line_ids[0].on_change_amount_currency()
```

Comment:
Got an error here when trying to create the planning lines

![image](https://user-images.githubusercontent.com/19620251/30743230-084db72c-9f9d-11e7-9cdc-4a824720f562.png)